### PR TITLE
API: allow to override rte image

### PR DESCRIPTION
**Pull request by fromanirh**
Add a field in our API to optionally allow to override the RTE image to deploy, mirroring wath the secondary-scheduler-operator does.
This gives us flexibiltiy at almost no cost, allowing easier test, enabling troubleshooting and hotfix delivery.

### CODE REVIEW

**Comment by yanirq** [comment]
Aren't we missing generated code here when it comes to API changes ?

**Comment by fromanirh** [comment]
> Aren't we missing generated code here when it comes to API changes ?

AFAIK no, because the client imports the API defs, so the changes are consumed that way.
xref: https://github.com/openshift-kni/numaresources-operator/blob/main/pkg/k8sclientset/generated/clientset/versioned/typed/numaresourcesoperator/v1alpha1/numaresourcesoperator.go#L24

File: controllers/numaresourcesoperator_controller.go [hunk 1]
```diff
@@ -210,6 +212,15 @@ func (r *NUMAResourcesOperatorReconciler) syncNUMAResourcesOperatorResources(ctx
 	return daemonSetsNName, nil
 }
 
+func (r *NUMAResourcesOperatorReconciler) getExporterImage(imgSpec string) string {
+	imageSpec := imgSpec
+	if imageSpec == "" {
+		imageSpec = r.ImageSpec
+	}
+	klog.InfoS("Using Exporter", "image", imageSpec)
```
Comment (Tal-or) [inline controllers/numaresourcesoperator_controller.go:220]: The log message isn't completely clear.
I would let the user know if we override the default image using the API filed or not

Comment (fromanirh) [reply controllers/numaresourcesoperator_controller.go:220]: ok, I'll revisit the log line.

**Comment by fromanirh** [comment]
/hold cancel
re-uploading remove the lgtm anyway

**Pull request merged**